src/relations/tls.py

```diff
@@ -217,21 +213,42 @@ def certificate_saved(self) -> bool:
             return False
         return self._relation.certificate_saved
 
+    @staticmethod
+    def _parse_tls_key(raw_content: str) -> str:
+        """Parse TLS key from plain text or base64 format."""
+        if re.match(r"(-+(BEGIN|END) [A-Z ]+-+)", raw_content):
+            return re.sub(
+                r"(-+(BEGIN|END) [A-Z ]+-+)",
+                "\n\\1\n",
+                raw_content,
+            )
+        return base64.b64decode(raw_content).decode("utf-8")
+
     def _on_set_tls_private_key(self, event: ops.ActionEvent) -> None:
         """Handle action to set unit TLS private key."""
         logger.debug("Handling set TLS private key action")
+        key = self._parse_tls_key(event.params["internal-key"])
+        if self.peer_unit_databag.private_key:
+            event.log("Warning: Deleted existing TLS private key")
+            logger.warning("Deleted existing TLS private key")
+        self.peer_unit_databag.private_key = key
+        event.log("Saved TLS private key")
+        logger.debug("Saved TLS private key")
```

Comment:
from the docs - https://charmhub.io/mysql/actions#get-password

> internal-key string
> The content of private key for internal communications with clients. Content will be auto-generated if this option is not specified.

Current logic is not generating a new private key if `internal-key` is not specified and the old private key (which should be rotated) is already stored in databag.